File: src/main/java/seremis/geninfusion/soul/gene/newAI/villagerMate/GeneAIVillagerMate.java
package seremis.geninfusion.soul.gene.newAI.villagerMate;

import seremis.geninfusion.api.soul.IAllele;
import seremis.geninfusion.api.soul.lib.Genes;
import seremis.geninfusion.soul.Gene;
import seremis.geninfusion.soul.MasterGene;
import seremis.geninfusion.soul.allele.AlleleBoolean;

public class GeneAIVillagerMate extends MasterGene {

    public GeneAIVillagerMate() {
        addControlledGene(Genes.GENE_AI_VILLAGER_MATE_INDEX);
    }

    @Override
    public Class<? extends IAllele> possibleAlleles() {
        return AlleleBoolean.class;
    }
}

File: src/main/java/seremis/geninfusion/soul/gene/newAI/GeneAIVillagerMate.scala
package seremis.geninfusion.soul.gene.newAI

import seremis.geninfusion.api.soul.EnumAlleleType
import seremis.geninfusion.api.soul.lib.Genes
import seremis.geninfusion.soul.MasterGene

class GeneAIVillagerMate extends MasterGene(EnumAlleleType.BOOLEAN) {
    addControlledGene(Genes.GENE_AI_VILLAGER_MATE_INDEX)
}